src/integration/cd/go/framework/ldap/LdapIntegrationTest.java

```diff
@@ -107,4 +118,60 @@ public void shouldSearchUsersFromMultipleSearchBases() throws Exception {
         assertThat(users, containsInAnyOrder(new User("pbanks", "P.Banks", "pbanks@example.com"), new User("sbanks", "S.Banks", "sbanks@example.com")));
         verify(ldapSpy).closeContextSilently(any(DirContext.class));
     }
+
+
+    @Test
+    public void search_shouldLimitTheSearchResults() throws Exception {
+        final LdapConfiguration ldapConfiguration = ldapConfiguration(new String[]{"ou=Employees,ou=Enterprise,ou=Principal,ou=system", "ou=Clients,ou=Enterprise,ou=Principal,ou=system"});
+        final UserMapper userMapper = ldapConfiguration.getUserMapper(new UsernameResolver());
+
+        final Ldap ldapSpy = spy(new Ldap(ldapConfiguration));
+
+        final List<User> allUsers = ldapSpy.search("(uid=*{0}*)", new String[]{"a"}, userMapper, Integer.MAX_VALUE);
+
+        assertThat(allUsers, hasSize(5));
+
+        final List<User> usersWhenMaxResultSpecified = ldapSpy.search("(uid=*{0}*)", new String[]{"a"}, userMapper, 3);
+
+        assertThat(usersWhenMaxResultSpecified, hasSize(3));
+        verify(ldapSpy, times(2)).closeContextSilently(any(DirContext.class));
+    }
+
+    @Test
+    public void search_shouldLimitTheSearchResultsAcrossSearchBases() throws Exception {
+        final LdapConfiguration ldapConfiguration = ldapConfiguration(new String[]{"ou=Employees,ou=Enterprise,ou=Principal,ou=system", "ou=Clients,ou=Enterprise,ou=Principal,ou=system"});
+        final UserMapper userMapper = ldapConfiguration.getUserMapper(new UsernameResolver());
+
+        final Ldap ldapSpy = spy(new Ldap(ldapConfiguration));
+
+        final List<User> allUsers = ldapSpy.search("(uid=*{0}*)", new String[]{"a"}, userMapper, Integer.MAX_VALUE);
+
+        assertThat(allUsers, hasSize(5));
+
+        final List<User> usersWhenMaxResultSpecified = ldapSpy.search("(uid=*{0}*)", new String[]{"a"}, userMapper, 4);
```

Comment:
It's not clear exactly how this is different from the previous test. I'm sure it has something to do with the data in the ldif file, but it can be more clear. Maybe with better variable names or better usernames?